mlog/stdlog.go

```diff
@@ -85,3 +85,13 @@ func (l *loggerWriter) Write(p []byte) (int, error) {
 	}
 	return len(p), nil
 }
+
+type panicLoggerWriter struct {
+	logFunc func(msg string, fields ...Field)
+}
+
+func (l *panicLoggerWriter) Write(p []byte) (int, error) {
+	trimmed := string(bytes.TrimSpace(p))
```

Comment:
For reference, can you share an example output? Curious how the trimming looks in practice.

Reply:
example output with current implementation:
```
{"level":"error","ts":1580823235.5002759,"msg":"panic: some comment about panic","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5003462,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500382,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500399,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5004318,"msg":"goroutine 43 [running]:","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500455,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500479,"msg":"main.(*Plugin).ServeHTTP(0xc000202190, 0xc00028a640, 0x18841c0, 0xc0002f5110, 0xc0001fe400)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5004952,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5005171,"msg":"/Users/shota/Work/Mattermost/github/mattermost-plugin-demo/server/http_hooks.go:22 +0x39","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500549,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500572,"msg":"github.com/mattermost/mattermost-server/plugin.(*hooksRPCServer).ServeHTTP(0xc0001e65e0, 0xc00036a960, 0x1d3afe8, 0x0, 0x0)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500585,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500603,"msg":"/Users/shota/go/pkg/mod/github.com/mattermost/mattermost-server@v1.4.1-0.20190911153151-98489b9e67d9/plugin/client_rpc.go:358 +0x2c3","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5006158,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500642,"msg":"reflect.Value.call(0xc0001ed200, 0xc0001f22b8, 0x13, 0x17761e5, 0x4, 0xc000278f18, 0x3, 0x3, 0x0, 0x0, ...)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500661,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5006819,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/reflect/value.go:460 +0x5f6","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500695,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500714,"msg":"reflect.Value.Call(0xc0001ed200, 0xc0001f22b8, 0x13, 0xc000292718, 0x3, 0x3, 0x0, 0x0, 0x0)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.50073,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500748,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/reflect/value.go:321 +0xb4","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500764,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.5007858,"msg":"net/rpc.(*service).call(0xc0001f83c0, 0xc000202410, 0xc0001f6360, 0xc0001f6370, 0xc000238900, 0xc0001f4940, 0x1631cc0, 0xc00036a960, 0x16, 0x1640bc0, ...)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500802,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.50082,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/net/rpc/server.go:377 +0x16f","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500833,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.50085,"msg":"created by net/rpc.(*Server).ServeCodec","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500862,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500879,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/net/rpc/server.go:474 +0x42b","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823235.500892,"msg":"","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
```

example output if we have output only lines starting in "panic":
```
{"level":"error","ts":1580823235.5002759,"msg":"panic: some comment about panic","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
```

Example output if we get rid of empty lines:
```
{"level":"error","ts":1580823751.69231,"msg":"panic: some comment about panic","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.6924129,"msg":"goroutine 54 [running]:","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692443,"msg":"main.(*Plugin).ServeHTTP(0xc000204190, 0xc0002c0820, 0x18841c0, 0xc0001d0e30, 0xc000322300)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692467,"msg":"/Users/shota/Work/Mattermost/github/mattermost-plugin-demo/server/http_hooks.go:22 +0x39","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692501,"msg":"github.com/mattermost/mattermost-server/plugin.(*hooksRPCServer).ServeHTTP(0xc0001f0440, 0xc0003827a0, 0x1d3afe8, 0x0, 0x0)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.69255,"msg":"/Users/shota/go/pkg/mod/github.com/mattermost/mattermost-server@v1.4.1-0.20190911153151-98489b9e67d9/plugin/client_rpc.go:358 +0x2c3","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692607,"msg":"reflect.Value.call(0xc0001e8c00, 0xc0001ee3b0, 0x13, 0x17761e5, 0x4, 0xc00006ff18, 0x3, 0x3, 0xc0000a2360, 0x0, ...)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692628,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/reflect/value.go:460 +0x5f6","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.6926458,"msg":"reflect.Value.Call(0xc0001e8c00, 0xc0001ee3b0, 0x13, 0xc00005ef18, 0x3, 0x3, 0x15537b6, 0xc00005ef00, 0xc00005eee8)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692666,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/reflect/value.go:321 +0xb4","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692686,"msg":"net/rpc.(*service).call(0xc0001f4500, 0xc0002c60a0, 0xc0001f82d0, 0xc0001f82e0, 0xc0001fe600, 0xc0001f0f40, 0x1631cc0, 0xc0003827a0, 0x16, 0x1640bc0, ...)","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692706,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/net/rpc/server.go:377 +0x16f","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692725,"msg":"created by net/rpc.(*Server).ServeCodec","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
{"level":"error","ts":1580823751.692788,"msg":"/usr/local/Cellar/go/1.13.4/libexec/src/net/rpc/server.go:474 +0x42b","plugin_id":"com.mattermost.demo-plugin","source":"plugin_stderr_panic"}
```

Probably we should pick last one? @lieut-data thoughts?

Reply:
@lieut-data kind reminder to give your advice on this :)

Reply:
Sorry for the delay on my part, @iomodo! I'm wondering if we should do without both the trimming and getting rid of empty lines. Odds are that we'll want to pipe these logs into a tool that dumps the original message, and it seems super handy to recover the original formatting intact in case /another/ tool expects it to be formatted in the same way.

It's perhaps slightly harder to consume in its raw JSON form, but I think it's a cost worth paying. Thoughts?

Reply:
Makes sense @lieut-data